src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java

```diff
@@ -479,14 +522,157 @@ void shouldDisplaySeriesPosition_withSeriesPositionPopulated_whenBookHasSeriesPo
     @Test
     void shouldNotDisplaySeriesPosition_whenBookDoesNotHaveSeriesPosition() {
         // given
-        bookForm = createBookForm(false);
+        bookForm = createBookForm(READ, false);
 
         bookForm.openForm();
 
         // then
         Assertions.assertFalse(bookForm.seriesPositionFormItem.isVisible());
     }
 
+    /**
+     * Tests whether a book is saved to the database when the save-event is called
+     */
+    @Test
+    void shouldSaveBookToDatabase() {
+        // given
+        bookForm = createBookForm(TO_READ, false);
+
+        // when
+        bookForm.addListener(BookForm.SaveEvent.class, event -> bookService.save(event.getBook()));
+        bookForm.saveButton.click();
+
+        // then
+        Assertions.assertEquals(1, bookService.count());
+        Book bookInDatabase = bookService.findAll().get(0);
+        assertBookInSpecificShelf(TO_READ, bookInDatabase);
+    }
+
+    /**
+     * Tests whether a book is updated in the database when the save-event is called
+     */
+    @Test
+    void shouldUpdateValuesForBook() {
+        // given
+        bookForm = createBookForm(TO_READ, false);
+        bookForm.addListener(BookForm.SaveEvent.class, event -> bookService.save(event.getBook()));
+        bookForm.saveButton.click();
+
+        Book savedBook = bookService.findAll().get(0);
+        populateBookFormWithExistingBook(READ, savedBook);
+
+        // when
+        bookForm.bookTitle.setValue("IT");
+        bookForm.authorFirstName.setValue("Stephen");
+        bookForm.authorLastName.setValue("King");
+        bookForm.bookGenre.setValue(Genre.HORROR);
+        bookForm.saveButton.click();
+
+        // then
+        Assertions.assertEquals(1, bookService.count());
+        Book updatedBook = bookService.findAll().get(0);
+        Assertions.assertEquals("IT", updatedBook.getTitle());
+        Assertions.assertEquals(READ, updatedBook.getPredefinedShelf().getPredefinedShelfName());
+        Assertions.assertEquals("Stephen", updatedBook.getAuthor().getFirstName());
+        Assertions.assertEquals("King", updatedBook.getAuthor().getLastName());
+        Assertions.assertEquals(Genre.HORROR, updatedBook.getGenre());
+        Assertions.assertEquals(dateStarted, updatedBook.getDateStartedReading());
+        Assertions.assertEquals(dateFinished, updatedBook.getDateFinishedReading());
```

Comment:
While it's great to have tests filled with books we like, could you change the title, author's first name and last name to use the respective global variables? It makes more sense to reuse those than creating string literals (both from a readability perspective and to prevent mistakes where we change one literal but not another occurrence of it somewhere else).

Reply:
It cannot use the global variables at that point because I have to use different values to make sure, that the book was in fact updated.
In "given" I create a book with the default values, add it to the database and then open the form to update the book. The form is populated with the information from the default book and I update these values with the stephen king book, asserting that these changes was saved to the database at the end.

I could create local variables in the given section of this test in order to avoid using just the strings. Something like
```java
String newTitle = "IT";
Author newAuthor = new Author("Stephen", "King");
```

Reply:
Ah, my bad. Local variables would be good.

Reply:
Could you also create a local variable for the new genre (`horror`)?

Reply:
done